coderd/files/cache.go

```diff
@@ -13,33 +13,43 @@ import (
 
 	archivefs "github.com/coder/coder/v2/archive/fs"
 	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/rbac"
+	"github.com/coder/coder/v2/coderd/rbac/policy"
 	"github.com/coder/coder/v2/coderd/util/lazy"
 )
 
+type AuthorizeFile func(ctx context.Context, subject rbac.Subject, action policy.Action, object rbac.Object) error
```

Comment:
nothing about this is "file" specific. it takes an `rbac.Object`. is this not a type def somewhere already? could we move this to a move general place with a more general name?

Reply:
True. Actually, I can just pass in an `rbac.Authorizer`. Not just the function.

https://github.com/coder/coder/pull/18348/commits/c9cf7809aab2e1904f3a9d2da7528d2e012f8eec

Reply:
Ahh I did it because it's an import loop

```
	imports github.com/coder/coder/v2/coderd/coderdtest from cache_internal_test.go

	imports github.com/coder/coder/v2/coderd from authorize.go
```

I fixed it because it was in a `internal` test, but I had to export the `cacheEntryValue`.

If we do our solution discussed in slack, this will be cleaned up.